python/pyarrow/parquet.py

```diff
@@ -2006,6 +2007,12 @@ def read_table(source, columns=None, use_threads=True, metadata=None,
         return dataset.read(columns=columns, use_threads=use_threads,
                             use_pandas_metadata=use_pandas_metadata)
 
+    warnings.warn(
+        "Passing 'use_legacy_dataset=True' to get the legacy behaviour is "
+        "deprecated as of pyarrow 8.0.0, and the legacy implementation will "
+        "be removed in a future version.",
+        DeprecationWarning, stacklevel=2)
```

Comment:
Do we want to use DeprecationWarning or FutureWarning?

Reply:
We can probably directly raise a FutureWarning, but all the others here are still raising DeprecationWarnings. Will do a separate PR to change those all from deprecation to future warning.